classes/submission/PKPAuthorDAO.inc.php

```diff
@@ -297,25 +298,23 @@ function deleteObject($author) {
 
 	/**
 	 * Delete an author by ID.
-	 * @param $authorId int
-	 * @param $submissionId int optional
+	 * @param $authorId int Author ID
+	 * @param $submissionId int Optional submission ID.
 	 */
 	function deleteById($authorId, $submissionId = null) {
 		$params = array((int) $authorId);
 		if ($submissionId) $params[] = (int) $submissionId;
-		$returner = $this->update(
```

Comment:
Removing this check could not leave orphaned author settings? Could happen that there's a database connection failure for the first query, but not for the second query?